io-engine/src/grpc/v0/mayastor_grpc.rs

```diff
@@ -1665,6 +1692,38 @@ impl mayastor_server::Mayastor for MayastorSvc {
         .await
     }
 
+    #[named]
+    async fn get_rebuild_history(
+        &self,
+        request: Request<RebuildHistoryRequest>,
+    ) -> GrpcResult<RebuildHistoryReply> {
+        let ctx = GrpcClientContext::new(&request, function_name!());
+        let args = request.into_inner();
+
+        self.serialized(ctx, args.uuid.clone(), false, async move {
+            trace!("{:?}", args);
+            let rx = rpc_submit::<_, _, nexus::Error>(async move {
+                let records = nexus_lookup(&args.uuid)?
+                    .rebuild_history()
+                    .unwrap()
+                    .iter()
+                    .map(RebuildHistoryRecord::from)
+                    .collect();
+                Ok(RebuildHistoryReply {
+                    nexus: String::from(
+                        nexus_lookup(&args.uuid)?.uuid().to_string(),
```

Comment:
I'm confused, why not simply: `nexus: args.uuid.clone()` ?

Reply:
changed.